### PR TITLE
feat(client): Report email opt-in status to firstrun page.

### LINKED ISSUE #3411: Inform the firstrun page if the user has clicked on the "email opt in" check box.
From https://bugzilla.mozilla.org/show_bug.cgi?id=1231496

--------------------------------

We want to see how many people opt-in to email communications as part of the Accounts signup within a GA dashboard.

Idea is to add a true/false parameter for the checkbox as part of the signup event. True = email opt-in is checked. False = email opt-in is not checked.

Is this possible?

--------------------------------

On the bedrock side, the change should be fairly trivial. However, we will need the status of that checkbox communicated to the page from the FxA iframe.

:stomlinson - What do you think? Would it be difficult to send an extra boolean along with the signup_must_verify event?

cc @jpetto, @vladikoff 

@vladikoff - do we send this info to GA now? I don't believe we do, but want to make sure.

### PR DESCRIPTION
**Pull request by shane-tomlinson**
Send `emailOptIn` with the `signup_must_verify` message.

`emailOptIn` can be `true` or `false`

fixes #3411

@vbudhram, @vladikoff and @jpetto - r?

@jpetto - rebuilding https://stomlinson.dev.lcip.org now with this change.

### CODE REVIEW

**Comment by philbooth** [comment]
@shane-tomlinson, I know you didn't ask me but r+ in case you want to merge this morning. Looks good to me! :grin:

File: app/tests/spec/models/auth_brokers/first-run.js [hunk 1]
```diff
@@ -97,15 +97,27 @@ define(function (require, exports, module) {
...
-            assert.isUndefined(result.halt);
+          .then(function (_result) {
+            result = _result;
           });
       });
+
+      it('notifies the iframe channel', function () {
+        assert.isTrue(
+          iframeChannel.send.calledWith(
+            broker._iframeCommands.SIGNUP_MUST_VERIFY, { emailOptIn: true }));
```
Comment (vbudhram) [inline app/tests/spec/models/auth_brokers/first-run.js]: Does `emailOptIn` default to false or true if not specified or is not boolean? Minor nit would be to add cases to cover these.

Comment (vbudhram) [inline app/tests/spec/models/auth_brokers/first-run.js]: Oops didn't see this was already merged :/

Comment (shane-tomlinson) [inline app/tests/spec/models/auth_brokers/first-run.js]: I can add more tests. :D

**Pull request merged**